src/Exp.js

```diff
@@ -399,9 +502,21 @@ class Exp {
                 siblingElement: expFor.nextElementSibling
             };
             if (arrayName in this.__storage.loopDefinitions) {
-                this.__storage.loopDefinitions[arrayName].push(expForInstance);
+                /* set siblingElement only if it exists and doesn't have exp-for */
+                const sibling = (expFor.nextElementSibling !== null && expFor.nextElementSibling.getAttribute('exp-for') !== null) ? null : expFor.nextElementSibling;
+                this.__storage.loopDefinitions[arrayName].push({
+                    template: template,
+                    parentElement: expFor.parentNode,
+                    siblingElement: sibling
+                });
             } else {
-                this.__storage.loopDefinitions[arrayName] = [expForInstance];
+                /* set siblingElement only if it exists and doesn't have exp-for */
+                const sibling = (expFor.nextElementSibling !== null && expFor.nextElementSibling.getAttribute('exp-for') !== null) ? null : expFor.nextElementSibling;
+                this.__storage.loopDefinitions[arrayName] = [{
```

Comment:
same comment as above: reuse `expForInstance` above from line 499. hence 506 line just modify `expForInstance[siblingElement] = ...` and 507 line just `push(expForInstance)`. also 499 line change from `const` to `let` or `var` so that it's mutable. all of this reason: code reuse and better clarity.